apps/site/src/pages/account/api-keys-panel/api-keys-list/shared/colored-card.tsx

```diff
@@ -0,0 +1,89 @@
+import { faClose } from "@fortawesome/free-solid-svg-icons";
+import { Box, BoxProps, IconButton } from "@mui/material";
+import { ReactNode } from "react";
+
+import { FontAwesomeIcon } from "../../../../../components/icons";
+
+const CloseCardButton = ({
+  onClick,
+  color,
+}: {
+  onClick: () => void;
+  color: string;
+}) => {
+  return (
+    <IconButton
+      data-testid="close-card-button"
+      onClick={onClick}
+      sx={{
+        position: "absolute",
+        top: 0,
+        right: 0,
+        "& svg": {
+          color,
+        },
+      }}
+    >
+      <FontAwesomeIcon icon={faClose} sx={{ fontSize: 16 }} />
+    </IconButton>
+  );
+};
+
+type Color = "purple" | "red";
+
+interface ColoredCardProps {
+  onClose?: () => void;
+  sx?: BoxProps["sx"];
+  children: ReactNode;
+  color?: Color;
+}
+
+const colorValues: Record<
```

Comment:
You'll probably need to define this within the component itself in order to access the theme palette (as I mentioned elsewhere that we probably don't want to use the string format for accessing colours)

Reply:
Moved it here https://github.com/blockprotocol/blockprotocol/pull/1257/commits/e2de1daeb0086a037932e3a21cbcd42aedf421fa